solutions/USACO open19-milk-factory.mdx

```diff
@@ -1,17 +1,103 @@
 
 [Official Analysis](http://www.usaco.org/current/data/sol_factory_bronze_open19.html)
 
 This problem can also be solved using [Depth-First Search](/silver/dfs) (a Silver topic). Represent the factory as a directed unweighted graph with edges from $b_i$ to $a_i$ for all $i$. For every node $i\in [1,N]$, we start a DFS at node $i$ and check whether this results in every other node being visited. We can keep track of whether the current DFS visits all nodes with a boolean array whose entries are initially set to `false`. If every node is visited, then we print $i$. Otherwise, if there exists no $i$ such that this condition is satisfied, then we print $-1$.
 
+## Implementation
+
 <LanguageSection>
+
+<JavaSection>
+	
+
+
+```java
+
+import java.io.*;
+import java.util.*;
+
+class Graph{
+    private int V;
+    private ArrayList<ArrayList<Integer> > adj;
+    Graph(int V){
+        this.V = V;
+        adj = new ArrayList<ArrayList<Integer> >(V);
+        for (int i = 0; i < V; i++){
+            adj.add(new ArrayList<Integer>()); // Creating the adjacency list
+        }
+    }
+    void addEdge(int u, int v) {
+
+        adj.get(v).add(u); // Connecting the two factories.
+
+    }
+    
+
+    int DFSUtil(int v,boolean visited[])
+    {
+        int ans  = 0;
+        // Mark the current node as visited
+        visited[v] = true;
+        ans++;
+
+        // Recur for all the vertices adjacent to this vertex
+        Iterator<Integer> i = adj.get(v).listIterator();
+        while (i.hasNext())
+        {
+            int n = i.next();
+            if (!visited[n]) {
+                ans += DFSUtil(n, visited); // recurse to the next edge/station in it's path and increase the number of stations visited.
+            }
+        }
+        return ans; // return number of visited stations.
+    }
+
+    // The function to do DFS traversal. It uses recursive DFSUtil()
+    int DFS(int v)
+    {
+        // Mark all the vertices as not visited(set as false by default)
+        boolean visited[] = new boolean[V];
+
+        // Call the recursive helper function to print DFS traversal
+        return DFSUtil(v, visited);
+    }
+}
+public class factory{
+
+    public static void main(String[] args) throws IOException {
+        Scanner sc = new Scanner(new File("factory.in"));
+        PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter("factory.out")));
+        int N = sc.nextInt(); Graph  g = new Graph(N); boolean used = false;
+        for (int i = 0; i < N - 1; i++) {
+            g.addEdge(sc.nextInt() - 1, sc.nextInt() - 1); // Meaning one can walk from station A to station B. 
+        }
+        for (int i = 0; i < N; i++) {
+            int ans = g.DFS(i);
+            if(ans == N) {
+                out.println(i + 1); // If the number of visited stations from the DFS is equal to the total number of stations then output. 
+                used = true;
+                break;
+            }
+        }
+        if(!used){
```

Comment:
bugs me that you don't seem to be consistent about spacing D:

`for (int i = 0; i < N; i++) {`

`if(ans == N) {`

`if(!used){`

```
while (i.hasNext())
        {
```

Why is there so much space in the `addEdge` function?

Reply:
@MrinallU I'll approve once you fix formatting and spacing consistency.